src/server.coffee

```diff
@@ -591,3 +597,24 @@ else
           type: 'restart-all'
       , 2000
     done()
+
+  # function to return process uptimes
+  exports.getUptime = (callback) ->
+
+    if cluster.isMaster
+      # send reponse back to API request
+      uptime =
+        master: process.uptime()
+      callback null, uptime
+    else
+      # send request to master
+      process.send
+        uptime: true
+
+      # listen for response from master
+      process.once 'message', (uptime) ->
```

Comment:
There is a minute chance that a server restart would be called while this code is waiting for a uptime response, this would cause it to get a restart message instead of a uptime response message. Its really unlikely so I don't think we have to change for using .once here but maybe we should add a .type property here to check that it is of the correct type before trying to call the callback?